src/Platform.Handlers/src/Xamarin.Platform.Handlers/App.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Xamarin.Platform.Hosting;
+
+namespace Xamarin.Platform
+{
+	public abstract class App : IApp
```

Comment:
Can this class have platform specific versions that all inherit from the application concept of that given platform?

For example 

```C#
	public abstract class App :

#if __ANDROID__
		global::Android.App.Application,
#elif __IOS__
		global::UIKit.UIApplicationDelegate,
#endif
		IApp
```

And then the user will just inherit from this App for MyApp and then that's the app that's also used to load up the application? 

I think in a perfect world there's only once concept of an "App"

Reply:
Yeah i was thinking something similar next, or this or native extension methods on the AppBuilder .. 

For example:
`.Init<TApplication>(global::Android.App.Application application)`
` .Init<TApplication>(global::UIKit.UIApplicationDelegate)`

Reply:
Fixed @PureWeen